Desktop/engine/enginesync.h

```diff
@@ -90,69 +104,84 @@ public slots:
+	bool		aChannelFree()						const;
+	bool		channelCooledDown(size_t channel)	const;
+
 #ifdef _WIN32 
 	void		fixPATHForWindows(QProcessEnvironment & env);
 #endif
 	
-	void		checkModuleWideCastDone();
-	void		resetModuleWideCastVars();
-	void		setModuleWideCastVars(Json::Value newVars);
-	bool		amICastingAModuleRequestWide()	{ return !_requestWideCastModuleJson.isNull(); }
 	size_t		maxEngineCount() const;
+	size_t		enginesIdleSoon() const;
 
 private slots:
 	void	deleteOrphanedTempFiles();
 	void	heartbeatTempFiles();
 
 	void	process();
 
-	void	moduleLoadingFailedHandler(		const QString & moduleName, const QString & errorMessage, int channelID);
-	void	moduleLoadingSucceededHandler(	const QString & moduleName, int channelID);
-	void	moduleUnloadingFinishedHandler(	const QString & moduleName, int channelID);
-
 	void	restartEngineAfterCrash(EngineRepresentation * engine);
-	void	restartKilledEngines();
+	void	restartKilledAndStoppedEngines();
 
-	void	logCfgReplyReceived(EngineRepresentation * engine);
+
+	void	logCfgReplyReceived(		EngineRepresentation * engine);
+	void	registerEngineForModule(	EngineRepresentation * engine, std::string modName);
+	void	unregisterEngineForModule(	EngineRepresentation * engine, std::string modName);
+	void	stopModuleEngine(			QString moduleName);
+	void	moduleInstallationFailedHandler(	const QString & moduleName, const QString & );
 	
 	void	maxEngineCountChanged();
-	void	startExtraEngine();
-	bool	anEngineIdleSoon();
+	void	startExtraEngines(size_t num=1);
+	bool	anEngineIdleSoon() const;
+	bool	moduleHasEngine(const std::string & name) { return _moduleEngines.count(name); }
+	void	resetListModel()	{ beginResetModel(); endResetModel(); } // lets keep things easy here, it doesnt have to be highperf
+
+	IPCChannel * channel(size_t channelNumber);
+
+private:
+	std::vector<EngineRepresentation *> orderedEngines() const;
 
 private:
 	static EngineSync				*	_singleton;
 	RFilterStore					*	_waitingFilter					= nullptr;
-
-	bool								_engineStarted					= false,
-										_filterRunning					= false;
+	bool								_filterRunning					= false,
+										_stopProcessing					= false;
 	int									_filterCurrentRequestID			= 0;
 	std::string							_memoryName,
-										_engineInfo,
-										_requestWideCastModuleName		= "";
-	Json::Value							_requestWideCastModuleJson		= Json::nullValue;
-	std::map<int, std::string>			_requestWideCastModuleResults;
+										_engineInfo;
 
 	std::queue<RScriptStore*>			_waitingScripts;
-	std::set<EngineRepresentation*>		_workers,		//Your typical run of the mill engine, can do analyses and Rscript/filter stuff
-										_rCmders,		//For those special occassions where you just want to shout at R in a more personal manner
-										_engines,		//Keeping track of all those engines isn't easy...
+	std::map<std::string,
+		EngineRepresentation * >		_moduleEngines;					///<An engine per module active. Engines will be started and closed as needed.
```

Comment:
I think you probably need one space after `///<` but I'm not sure if it's required. I would argue that it's more readable with the space.